First commit with refactoring

kakavlya opened the pull request.

kakavlya committed:
First commit with refactoring

kakavlya committed:
Revert "First commit with refactoring"

This reverts commit a03a0ba7c48a76a420ac92a0151fe71aefbc6687.

kakavlya committed:
Revert "Revert "First commit with refactoring""

This reverts commit e12ecd1ab7c08299d72e3e7c438bf9f3b34bf5fb.

kakavlya committed:
Revert "Revert "Revert "First commit with refactoring"""

This reverts commit fce8caac117119d044b588631f927c0aed504402.

kakavlya committed:
Revert "Revert "Revert "Revert "First commit with refactoring""""

This reverts commit 8009daed9c498e8f1aa4be22fbf781d4d442fa54.

HolyMonkey commented on Assets/Level/Scripts/Enemy/Enemy.cs:
```diff
@@ -0,0 +1,20 @@
+
+    private void Start()
+    {
+        _target = Random.insideUnitCircle * 4;
```
дубляж кода

HolyMonkey replied:
+ магическое число

HolyMonkey commented on Assets/Level/Scripts/General/GameController.cs:
```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GameController : MonoBehaviour
```
Ну да, контроллер

HolyMonkey commented on Assets/Level/Scripts/General/GameController.cs:
```diff
@@ -0,0 +1,36 @@
+
+public class GameController : MonoBehaviour
+{
+    public static GameController Constroller;
```
Мы не используем синглтон и это корявая реализация

HolyMonkey commented on Assets/Level/Scripts/General/GameController.cs:
```diff
@@ -0,0 +1,36 @@
+public class GameController : MonoBehaviour
+{
+    public static GameController Constroller;
+    private const string Message = "SendMEssage";
```
SendMEssage

HolyMonkey commented on Assets/Level/Scripts/General/GameController.cs:
```diff
@@ -0,0 +1,36 @@
+        _target.SetActive(true);
+    }
+
+    private void Update(){
```
как курица лапой

HolyMonkey commented on Assets/Level/Scripts/Player/Player.cs:
```diff
@@ -0,0 +1,53 @@
+    private void Update()
+    {
+        if (_timePassed)
+        {
+            _elapsedTime -= Time.deltaTime;
+            if (_elapsedTime < 0)
+            {
+                _timePassed = false;
+                _playerMover.Speed /= 2;
+            }
+        }
```
Должна быть возможность подобрать несколько бустеров за раз

HolyMonkey replied:
Всё ешё актуально

HolyMonkey commented on Assets/Level/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,16 @@
+{
+    [SerializeField] private float _speed;
+
+    public float Speed { get => _speed; set => _speed = value; }
```
зачем? Чисто публичное поле

HolyMonkey commented on Assets/Level/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,16 @@
+
+    private void Update()
+    {
+        Vector3 movement = new Vector3(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"), 0);
```
https://trello.com/c/UAeMGORw/17-%D0%BE%D1%82%D0%BA%D0%B0%D0%B7-%D0%BE%D1%82-%D1%81%D1%82%D1%80%D0%BE%D0%BA%D0%BE%D0%B2%D1%8B%D1%85-%D0%BB%D0%B8%D1%82%D0%B5%D1%80%D0%B5%D0%BB%D0%BE%D0%B2

kakavlya committed:
Removed singleton and made fixes according to explanation.

   [SerializeField] private float _speed;

    public float Speed { get => _speed; set => _speed = value; }
Оставил как есть, так как через  SerializedField устанавливаем _speed в Unity. Через гетер и сетер модифицируем скорость. Если это не правильно решение, какое лучше использовать ?

HolyMonkey commented on Assets/Level/Scripts/Enemy/Enemy.cs:
```diff
@@ -0,0 +1,26 @@
+        SetTarget();
+    }
+
+    private void SetTarget()
```
Set должен устанавливать состояние из параметра. Почитай про CQRS

HolyMonkey commented on Assets/Level/Scripts/General/Game.cs:
```diff
@@ -0,0 +1,30 @@
+        _endObject.SetActive(true);
+    }
+
+    private void Update(){
+        foreach (var enemy in _enemies)
+        {
+            if (enemy == null)
+                continue;
+
+                if (Vector3.Distance(_player.gameObject.gameObject.GetComponent<Transform>().position, enemy.gameObject.gameObject.transform.position) < 0.2f)
+                {
+                    _player.SendMessage(Message, enemy);
+                }
+        }
+    }
```
Что с форматированием?

HolyMonkey commented on Assets/Level/Scripts/General/Game.cs:
```diff
@@ -0,0 +1,30 @@
+            if (enemy == null)
+                continue;
+
+                if (Vector3.Distance(_player.gameObject.gameObject.GetComponent<Transform>().position, enemy.gameObject.gameObject.transform.position) < 0.2f)
```
Прочитай ещё раз эту строку
